File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        lookup = {0: 1}
        array = [x for x in nums]
        curr_sum = 0
        output = 0

        def isOdd(val):
            return val % 2 != 0

        for i in range(len(nums)):
            if (isOdd(nums[i])):
                array[i] = 1
            else:
                array[i] = 0

        for i in range(len(nums)):
            curr_sum += array[i]
            prefix = curr_sum - k

            if (prefix in lookup):
                output += lookup[prefix]
            if (curr_sum not in lookup):
                lookup[curr_sum] = 0
            lookup[curr_sum] += 1

        return output
            
        

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.js
var numberOfSubarrays = function(nums, k) {
    for (let i = 0; i < nums.length; i++) {
        if (isOdd(nums[i])) nums[i] = 1
        else nums[i] = 0
    }

    function isOdd(num) {
        return num % 2 !== 0
    }
    let currentSum = 0
    let lookup = {0: 1}
    let output = 0

    for (let i = 0; i < nums.length; i++) {
        currentSum += nums[i]

        if ((currentSum - k) in lookup) {
            output += lookup[currentSum - k]
        }
        if (!(currentSum in lookup)) lookup[currentSum] = 0
        lookup[currentSum] += 1
    }
    return output

};